### PR TITLE
feat(cmd/helm): add '--password-stdin' option where '--password' is allowed

### PR DESCRIPTION
**Pull request by glb**
Mirrors similar behaviour in the Docker CLI client, minus the warning that providing passwords on the command line is insecure.

Related: #3206, #4723 (kind of).

### CODE REVIEW

**Comment by glb** [comment]
It seems like there's a real opportunity here to refactor credential handling in the commands, but I don't know the code base well enough to do it yet. Happy to take direction here.

File: cmd/helm/inspect.go
Comment (bacongobbler) [inline cmd/helm/inspect.go]: could we refactor all this into a general function? Something like

```
func readPassword(username, password string, readPasswordFromStdin bool) (string, error) {
}
```

That way we're not duplicating the same code in 5 different places, and the code in each command can just be

```
password, err := readPassword(insp.username, insp.password, insp.passwordStdin)
if err != nil {
    return err
}
```

Which makes maintenance for this particular feature *much* more manageable.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: cmd/helm/inspect.go
Comment (glb) [reply cmd/helm/inspect.go]: Definitely; I was thinking along similar lines but wasn't sure where to put the new function -- currently leaning towards a new file. I'll think of a first draft location and will probably make it take a `Reader` as well so that I can write some unit tests for it.

Comment (bacongobbler) [reply cmd/helm/inspect.go]: In general we've been placing helper functions under `cmd/helm/helm.go` but if it makes sense to go under `pkg/` then perhaps creating a new package called `pkg/password`? Up to you.

Comment (glb) [reply cmd/helm/inspect.go]: OK, thanks! I'll likely keep it internal to start, so `cmd/helm/helm.go` seems like the preferred choice at the moment. I'll take a shot at this later today.

Comment (glb) [reply cmd/helm/inspect.go]: Update: I did make an update but have forgotten the magic I needed to do in order to do a build; the `glide` command is running out of file descriptors. I remember it was something to do with `docker --ulimit nofiles=1000000` but still ran into an error. Will try building outside a container tomorrow.

Also: Is there an explicit reason why the `helm inspect` command and its children didn't use `PersistentFlags`? There seems to be a bunch of code that could be eliminated by using it. Perhaps I'll find out when I get the build working...

**Comment by bacongobbler** [comment]
Thanks @glb. Would it be all right if the `helm inspect` refactor was split out into a separate PR? It looks like there might've been some changes to how the flags are organized which we may need to take a closer look at, but it shouldn't block the `--password-stdin` flag from going through.

**Comment by glb** [comment]
Sure, no worries at all @bacongobbler ... I'll split the changes out when I write the unit tests tomorrow. Thanks for taking a look!

**Comment by glb** [comment]
Added unit tests

**Comment by glb** [comment]
@bacongobbler how does this look? I'll admit that I'm now a bit uncertain about whether we need an explicit `password-stdin` flag; `repo-add` already prompted for the password if `username` was provided but `password` was not, and I've kept that behaviour. The only benefit that `password-stdin` has is it makes the user's intent clear and might surface the capability more clearly.

I'm OK with either way -- it would be very simple to remove the `password-stdin` flag and update the tests.

**Comment by bacongobbler** [comment]
Traveling to Europe/weekending, sorry. I'll get back to this on Monday. :)

**Review by bacongobbler — APPROVED** [review]
LGTM but since this is a `size/L` it'll need to be signed off by another maintainer before we can merge. Thanks for refactoring this!

**Comment by Carl5163** [comment]
Have there been any updates on getting another reviewer for this? It would be very helpful if possible.

**Comment by glb** [comment]
@Carl5163 I had somewhat moved on as this didn't seem to be getting much traction; I _may_ have some time to rebase / re-work this PR (again) next week.

@bacongobbler would there be any point to trying again with this PR or is the team heads-down on Helm 3?

**Comment by bacongobbler** [comment]
The team is mostly heads-down on Helm 3. If I had more time in the world I'd love to take the time and review this, but unfortunately I don't have the time to work on both Helm 2 and Helm 3 work. I'd ask around and see if there are other maintainers that have the time to review. Sorry!

**Comment by bacongobbler** [comment]
Hey @glb, sorry for not getting back to you sooner. The team has been focusing for the past few months on getting Helm 3 out the door, and that has meant it's been harder to find time to review feature additions for Helm 2.

Helm 2.15 is soon to be released, and will be the last feature release for Helm 2. Because of the time commitment required to rebase this and test it, we won't be able to find the time to review this. That being said, if you want to try again by implementing this for Helm 3, that would be much appreciated. I'd hold off for a few weeks until 2.15 is out as we are planning to move the `dev-v3` branch over to be the new `master` branch, so there will be some shuffling around for a bit, but once that's done we'd be happy to consider this for the 3.1.0 release.

Thanks again!

**Pull request closed without merging**
